### PR TITLE
TNT-38923 - get_offers implementation

### PR DESCRIPTION
**Pull request by dcottingham**
Usage:  client.get_offers(options)
options.request is a DeliveryRequest object, which is generated by the openapi tool, and includes all the data to send to Delivery API.  Generated Delivery API functions do conversion from DeliveryRequest -> json on the request side, and json -> DeliveryResponse on the response side.  A response envelope which includes DeliveryResponse as well as other metadata is returned to user.  By default the call is done synchronously, but if options.callback is provided then the call is done asynchronously.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -13,4 +13,6 @@ current_version = 0.0.1
 commit = True
 tag = True
 [bumpversion:file:setup.cfg]
+search = target_python_sdk=={current_version}
+replace = target_python_sdk=={new_version}
```
Comment (mptap) [inline setup.cfg:18]: Nice to have this automated!

File: delivery_api_client/api/__init__.py [hunk 2]
```diff
@@ -3,4 +3,4 @@
 # flake8: noqa
 
 # import apis into api package
-from openapi_client.api.delivery_api import DeliveryApi
+from delivery_api_client.api.delivery_api import DeliveryApi
```
Comment (mptap) [inline delivery_api_client/api/__init__.py:6]: I'm curious to know why it was called `openapi_client` earlier instead of `delivery_api_client`.

Comment (dcottingham) [reply delivery_api_client/api/__init__.py:6]: That was just the default name given by the codegen tool (see codegeneration/package.json), but changed the name to be more in line with what our other SDKs use.

File: delivery_api_client/api/delivery_api.py [hunk 3]
```diff
@@ -68,35 +76,61 @@ def execute_with_http_info(self, ims_org_id, session_id, delivery_request, **kwa
...
-        all_params.append('_return_http_data_only')
-        all_params.append('_preload_content')
-        all_params.append('_request_timeout')
+        all_params = [
+            'ims_org_id',
+            'session_id',
+            'delivery_request',
+            'version'
+        ]
+        all_params.extend(
```
Comment (5amfung) [inline delivery_api_client/api/delivery_api.py:122]: Can this be one statement [a, b, c, ...]?  Curious why we have to call extend?

File: target_python_sdk/tests/responses/prefetch.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            "type":"actions",
+            "content":[{
+               "type":"html",
+               "content": "<p> Home Page </p>",
+               "selector": "#parent"
+            }]
+         }]
+      }]
+   }
+}
```
Comment (5amfung) [inline target_python_sdk/tests/responses/prefetch.json:52]: Missing EOF newline.

File: target_python_sdk/tests/responses/execute.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            "options": [{
+              "content": "<p>Hello World</p>",
+              "type": "html",
+              "eventToken":
+                "hi+ys8FLH8AC6VEf02b7k5NWHtnQtQrJfmRrQugEa2qCnQ9Y9OaLL2gsdrWQTvE54PwSz67rmXWmSnkXpMMM2Q=="
+            }]
+         }
+      ]
+   }
+}
```
Comment (5amfung) [inline target_python_sdk/tests/responses/execute.json:60]: Missing EOF newline.

File: target_python_sdk/tests/responses/invalid_request_error.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "message": "Invalid request: Invalid imsOrg -  bad",
+  "status": 400
+}
```
Comment (5amfung) [inline target_python_sdk/tests/responses/invalid_request_error.json:4]: Missing EOF newline.

File: target_tools/networking.py.bck [hunk 7]
```diff
@@ -11,7 +11,7 @@
 """Networking related functions"""
 import requests
 
-
+# TODO - this might be specific to JS
```
Comment (5amfung) [inline target_tools/networking.py.bck:14]: Is this file `networking.py.bck` included by mistake?

File: target_python_sdk/tests/responses/default.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+            "options": [{
+              "content": "<p>Hello World</p>",
+              "type": "html",
+              "eventToken":
+                "hi+ys8FLH8AC6VEf02b7k5NWHtnQtQrJfmRrQugEa2qCnQ9Y9OaLL2gsdrWQTvE54PwSz67rmXWmSnkXpMMM2Q=="
+            }]
+         }
+      ]
+   }
+}
```
Comment (5amfung) [inline target_python_sdk/tests/responses/default.json:43]: Missing EOF newline.

File: target_python_sdk/tests/responses/customer_ids.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            "options": [{
+              "content": "<p>Hello World</p>",
+              "type": "html",
+              "eventToken":
+                "hi+ys8FLH8AC6VEf02b7k5NWHtnQtQrJfmRrQugEa2qCnQ9Y9OaLL2gsdrWQTvE54PwSz67rmXWmSnkXpMMM2Q=="
+            }]
+         }
+      ]
+   }
+}
```
Comment (5amfung) [inline target_python_sdk/tests/responses/customer_ids.json:48]: Missing EOF newline.

**Review by 5amfung — APPROVED** [review]
A few minor issues otherwise LGTM.

**Review by mptap — APPROVED** [review]
LGTM, apart from missing newlines EOF in test fixtures

File: target_tools/networking.py.bck (hunk 7 above)
Comment (dcottingham) [reply target_tools/networking.py.bck:14]: good catch - removed

File: delivery_api_client/api/delivery_api.py (hunk 3 above)
Comment (dcottingham) [reply delivery_api_client/api/delivery_api.py:122]: Everything under the /delivery_api_client directory is generated by the openapi generator tool.  So I don't modify anything in there.

**Pull request merged**